benchmark/single-source/FlattenDistanceFromTo.swift

```diff
@@ -0,0 +1,58 @@
+//===--- FlattenDistanceFromTo.swift --------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+public let FlattenDistanceFromTo = [
+  BenchmarkInfo(
+    name: "FlattenDistanceFromTo.RandomAccess.16x16",
+    runFunction: { withRandomAccess(16, 16, $0) },
+    tags: [.validation, .api]),
+  
+  BenchmarkInfo(
+    name: "FlattenDistanceFromTo.RandomAccess.16x32",
+    runFunction: { withRandomAccess(16, 32, $0) },
+    tags: [.validation, .api]),
+  
+  BenchmarkInfo(
+    name: "FlattenDistanceFromTo.RandomAccess.32x16",
+    runFunction: { withRandomAccess(32, 16, $0) },
+    tags: [.validation, .api]),
+  
+  BenchmarkInfo(
+    name: "FlattenDistanceFromTo.RandomAccess.32x32",
+    runFunction: { withRandomAccess(32, 32, $0) },
+    tags: [.validation, .api]),
+]
+
+@inline(never)
+public func withRandomAccess(
+  _ outer: Int,
+  _ inner: Int,
+  _ iterations: Int
+) {
+  var value = 0 as Int
+  let minor = repeatElement(00000, count: inner)
+  let major = repeatElement(minor, count: outer)
+  let flattened: FlattenSequence = major.joined()
```

Comment:
These could be at the top level, so that creating them isn't part of the measurement.

Reply:
Are you sure you want to move these? It would be a lot of ceremony for `3` integers. Note that `distance(from:to:)` is called `2 * outer^2 * inner^2 * iterations` times. I'm asking just in case you thought they were arrays.

```
▿ Swift.FlattenSequence<Swift.Repeated<Swift.Repeated<Swift.Int>>>
 ▿ _base: Swift.Repeated<Swift.Repeated<Swift.Int>>
  - count: 32
  ▿ repeatedValue: Swift.Repeated<Swift.Int>
   - count: 32
   - repeatedValue: 0
```

Reply:
Nevermind. I found a solution I don't hate, then added a few sequential benchmarks for good measure. It's boilerplate-y but symmetrical. You could make it generic, but perhaps manual specialization is preferred in benchmarks?